dirsrvtests/tests/suites/ldapi/ldapi_test.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import pytest
+import os
+import subprocess
+import grp
+import time
+from pwd import getpwnam
+from lib389._constants import DEFAULT_SUFFIX
+from lib389._mapped_object import DSLdapObject
+from lib389.idm.user import UserAccounts
+from lib389.topologies import topology_st as topo
+
+log = logging.getLogger(__name__)
+
+
+def test_ldapi_authdn_attr_rewrite(topo, request):
+    """Test LDAPI Authentication DN mapping feature
+
+    :id: e8d68979-4b3d-4e2d-89ed-f9bad827718c
+    :setup: Standalone Instance
+    :steps:
+        1. Set LDAPI configuration
+        2. Create LDAP user
+        3. Create OS user
+        4. Create entries under cn=config for auto bind subtree and mapping entry
+        5. Do an LDAPI ldapsearch as the OS user
+        6. OS user was mapped expected LDAP entry
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+        5. Success
+        6. Success
+    """
+
+    LINUX_USER = "ldapi_test_lib389_user"
+    LINUX_PWD = "5ecret_137"
+    AUTH_DN_ATTR = "authenticateAsDN"
+    LDAP_ENTRY_DN = "uid=test_ldapi,ou=people,dc=example,dc=com"
+
+    def fin():
+        # Remove the OS user
+        try:
+            subprocess.run(['userdel', LINUX_USER])
+        except:
+            pass
+    request.addfinalizer(fin)
+
+    # Must be root
+    if os.geteuid() != 0:
+        return
+
+    # Perform config tasks
+    topo.standalone.config.set('nsslapd-accesslog-logbuffering', 'off')
+    topo.standalone.config.set('nsslapd-ldapiautoauthdnattr', AUTH_DN_ATTR)
+    topo.standalone.config.set('nsslapd-ldapientrysearchbase', 'cn=auto_bind,cn=config')
+    topo.standalone.config.set('nsslapd-ldapimaptoentries', 'on')
+    topo.standalone.config.set('nsslapd-ldapiuidnumbertype', 'uidNumber')
+    topo.standalone.config.set('nsslapd-ldapigidnumbertype', 'gidNumber')
+    ldapi_socket_raw = topo.standalone.config.get_attr_val_utf8('nsslapd-ldapifilepath')
+    ldapi_socket = ldapi_socket_raw.replace('/', '%2F')
+
+    # Create LDAP user
+    users = UserAccounts(topo.standalone, DEFAULT_SUFFIX)
+    user_properties = {
+        'uid': 'test_ldapi',
+        'cn': 'test_ldapi',
+        'sn': 'test_ldapi',
+        'uidNumber': '2020',
+        'gidNumber': '2020',
+        'userpassword': 'password',
+        'description': 'userdesc',
+        'homeDirectory': '/home/test_ldapi'}
+    users.create(properties=user_properties)
+
+    # Create OS user
+    subprocess.run(['useradd', '-p', LINUX_PWD, LINUX_USER ])
+
+    # Get UID GID
+    uid = getpwnam(LINUX_USER)[2]
+    gid = grp.getgrnam(LINUX_USER)[2]
+
+    # Create subtree to store LDAPI mappings under cn=config
+    config = DSLdapObject(topo.standalone, dn="cn=auto_bind,cn=config")
```

Comment:
If this is could to be an auto_bind uid/gid mapping type we should probably give it a real objectClass and a dsldapobject subclass.

Reply:
Perhaps the Config class could be extended so we can add entries to it instead of creating a new class?  

And yes, I can add a new objectclass/attributes for these config entries under "cn=auto_bind,cn=config".

Reply:
Nah, we need a new dsldapobject class so we can expose search/list/create/get from the cli. And that also is part of why an oc is needed to gen the correct filters for this.